src/cli.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+from pathlib import Path
+from .types import Command
+
+
+class Colors:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+    MAGENTA = "\u001b[35m"
+
+
+def error(msg: str) -> None:
+    print(f"{Colors.FAIL}[ERROR]{Colors.ENDC} {msg}")
+    exit(1)
+
+
+def warn(msg: str, ret_only: bool = False) -> str:
+    msg = f"{Colors.WARNING}[WARNING]{Colors.ENDC} {msg}"
+    if not ret_only:
+        print(msg)
+    return msg
+
+
+def success(msg: str) -> None:
+    print(f"{Colors.OKGREEN}[SUCCESS]{Colors.ENDC} {msg}")
+
+
+def info(msg: str) -> None:
+    print(f"{Colors.OKBLUE}[INFO]{Colors.ENDC} {msg}")
+
+
+def todo(msg: str) -> None:
+    print(f"{Colors.MAGENTA + Colors.BOLD}[TODO]{Colors.ENDC} {msg}")
+
+
+def validate_path_from_cli(
+    _path: str, argument: str, allow_mkdir: bool = False
+) -> Path:
+    """
+    Allow argparse to actually validate -p/--path-to-linux argument and
+    provide helpful feedback
+    """
+    try:
+        posix_path = Path(_path.strip())
+    except:
```

Comment:
Is there a more specific exception type you can catch here?

Reply:
I've opted to remove this try/except completely.

Nathan mentioned this earlier too: https://github.com/ClangBuiltLinux/misc-scripts/pull/2#discussion_r1269701328

It is, in fact, pretty impossible to conjure up a string that pathlib can't convert to a path.